report_xlsx/controllers/main.py

```diff
@@ -0,0 +1,104 @@
+# Copyright (C) 2017 Creu Blanca
+# Copyright 2021 ForgeFlow S.L.
+# License AGPL-3.0 or later (https://www.gnuorg/licenses/agpl.html).
+
+import json
+import logging
+
+from werkzeug.urls import url_decode
+
+from odoo.http import (
+    content_disposition,
+    request,
+    route,
+    serialize_exception as _serialize_exception,
+)
+from odoo.tools import html_escape
+from odoo.tools.safe_eval import safe_eval, time
+
+from odoo.addons.web.controllers import report
+
+_logger = logging.getLogger(__name__)
+
+
+class ReportController(report.ReportController):
+    @route()
+    def report_routes(self, reportname, docids=None, converter=None, **data):
+        if converter == "xlsx":
+            report = request.env["ir.actions.report"]._get_report_from_name(reportname)
+            context = dict(request.env.context)
+            if docids:
+                docids = [int(i) for i in docids.split(",")]
+            if data.get("options"):
+                data.update(json.loads(data.pop("options")))
+            if data.get("context"):
+                data["context"] = json.loads(data["context"])
+                context.update(data["context"])
+            xlsx = report.with_context(**context)._render_xlsx(docids, data=data)[0]
```

Comment:
If now the _render_xlsx method needs 2 more parameters data.

```
@api.model
def _render_xlsx(self, report_ref, docids, data):
```
I suppose it will be necessary to add the name of the report. no?

```
xlsx = report.with_context(**context)._render_xlsx(
                reportname, docids, data=data)[0]
```
this now fails:

![Selección_035](https://user-images.githubusercontent.com/6359121/194329181-2efd9217-662a-494a-a2f0-1c17bd7d29d9.jpg)